draft-ietf-dnsop-dns-capture-format-00.md

```diff
@@ -233,32 +233,32 @@ that the files produced can be subject to further compression using general purp
+
+    * Any structured capture format that does not capture the DNS payload byte for byte will be limited to some extent in
+      that it cannot represent "malformed" DNS packets. Only those packets that can be transformed reasonably into the
+      structured format can be represented by the format. So if a query is malformed this will lead to the (well formed) DNS responses
+      with error code FORMERR appearing as "unmatched".
+
+    * There are two distinct types of packets that are considered "malformed". These are packets which can be decoded sufficiently
+      to allow matching with a corresponding query or response ("partially malformed"), and those that cannot ("completely malformed").
+      To be matched, it must be possible to decode the Primary ID (see (#primary-id)), and, if the DNS message is a query, the Secondary ID
+      (see (#secondary-id)). In other words, it must be possible to decode the IP source and destination addresses, the UDP or
+      TCP source and destination ports, the DNS header ID, the DNS header flags/opcode/rcode fields, and, if the DNS message is a query,
+      the entire DNS header and the first question section entry, which implies that in a DNS query QDCOUNT must be at least 1 or the
+      packet is completely malformed.
+
+    * When processing a partially malformed DNS message, the generated Q/R data item must be populated with the information
+      in the Primary ID, the Secondary ID (if relevant) and the packet timestamp. All other fields in the Q/R data item are
+      optional, and so may be omitted. Implementations may choose to include individual fields if confident that the field
+      content is correctly decoded.
 
 Rationale: Many name servers will process queries on a best-effort basis in accordance with Postel's Law, and do not insist on
-completely well-formed packets. Name servers will also generally ignore any trailing data following well-formed DNS content. Users may wish
-to be informed of such transactions, or input data that cannot be decoded to even a DNS header and which therefore cannot be meaningfully
-processed as part of the query/response stream, and may wish to be able to analyse these malformed inputs as, for example, possible attack
-vectors. Therefore these interactions with the name server should be recorded where possible, but flagged as malformed.
-
-QUESTION: Should a valid DNS header include additional conditions?
-
-* The flags and codes words has valid values for operation code and response code.
-* A query has response code 0.
-* All zero bits are set to 0.
+completely well-formed packets. Users may wish to be informed of such transactions, or input data that cannot be decoded to even
+a DNS header and which therefore cannot be meaningfully processed as part of the query/response stream, and may wish to be able to
+analyse these malformed inputs. Therefore these interactions with the name server should be recorded where possible. but flagged as malformed.
```

Comment:
It should be be possible to optionally record these messages directly in the C-DNS format.